src/main/scala/mapreduce/FrequencyHistogram.scala

```diff
@@ -0,0 +1,83 @@
+package mapreduce
+
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import akka.pattern.{ask, pipe}
+import akka.util.Timeout
+
+import java.io.{BufferedWriter, File, FileWriter}
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.io.{BufferedSource, Source}
+import scala.language.postfixOps
+
+case object ProcessFile
+case class CountedLine(words: mutable.Map[String, Int])
+case object WriteResult
+case object GetResult
+case class Result(histogram: mutable.Map[String, Int])
+
+class MapActor extends Actor {
+  def receive = {
+    case line: String =>
+      val words = mutable.TreeMap[String, Int]()
+      line.split("[\\s.,:;!?()*'\"\\[\\]]+").map(_.toLowerCase())
+        .foreach(word => words.updateWith(word) {
+          case Some(count) => Some(count + 1)
+          case None => Some(1)
+        })
+      sender ! CountedLine(words)
+  }
+}
+
+class ReduceActor extends Actor {
+  val result = new mutable.TreeMap[String, Int]()
+  def receive = {
+    case CountedLine(words) =>
+      words.foreach(word => result.updateWith(word._1) {
+        case Some(count) => Some(count + word._2)
+        case None => Some(word._2)
+      })
+    case GetResult => sender ! Result(result)
```

Comment:
Try implementing push approach not pulling the result. So, as you have the result give it back to the MapReduce actor.